formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt

```diff
@@ -101,6 +101,19 @@ internal open class StreamingJsonDecoder(
         }
     }
 
+    private fun SerialDescriptor.getJsonElementIndex(key: String): Int {
+        val index = this.getElementIndex(key)
+        if (!json.configuration.useAlternativeNames) return index
+        // Fast path, do not go through ConcurrentHashMap.get
+        // Note, it blocks ability to detect collisions between the primary name and alternate,
+        // but it eliminates a significant performance penalty (about -15% without this optimization)
+        if (index != UNKNOWN_NAME) return index
```

Comment:
Performance nit: this if should go before `json.configuration.useAlternativeNames`

Reply:
Not sure if this would make any difference, since `if (!json.configuration.useAlternativeNames)` is a very predictable check, but ok